etna/metrics/base.py

```diff
@@ -146,29 +146,38 @@ def _validate_segment_columns(y_true: TSDataset, y_pred: TSDataset):
             )
         for segment in segments_true:
             for name, dataset in zip(("y_true", "y_pred"), (y_true, y_pred)):
-                if "target" not in dataset.loc[:, segment].columns:
+                if (segment, "target") not in dataset.columns:
                     raise ValueError(
                         f"All the segments in {name} should contain 'target' column. Segment {segment} doesn't."
                     )
 
     @staticmethod
-    def _validate_timestamp_columns(timestamp_true: pd.Series, timestamp_pred: pd.Series):
+    def _validate_timestamp_columns(y_true: TSDataset, y_pred: TSDataset):
         """
-        Check that ``y_true`` and ``y_pred`` have the same timestamp.
+        Check that ``y_true`` and ``y_pred`` have the same timestamps.
 
         Parameters
         ----------
-        timestamp_true:
-            y_true's timestamp column
-        timestamp_pred:
-            y_pred's timestamp column
+        y_true:
+            y_true dataset
+        y_pred:
+            y_pred dataset
 
         Raises
         ------
         ValueError:
             If there are mismatches in ``y_true`` and ``y_pred`` timestamps
         """
-        if set(timestamp_pred) != set(timestamp_true):
+        df_true = y_true.df.loc[:, pd.IndexSlice[:, "target"]].sort_index(axis=1)
+        df_pred = y_pred.df.loc[:, pd.IndexSlice[:, "target"]].sort_index(axis=1)
+
+        df_true_isna = df_true.isna()
```

Comment:
Why can't we just ckech that both of `df_true_isna` and `df_pred_isna` sum to 0? As I understand we also need to compare the index here, does `equals` do it?

Reply:
I'm not sure that your suggested solution gives the same result as the initial solution. In initial solution we select segment from `ts`, it uses `first_valid_index` under the hood and skips the first NaNs.

Here we apply `first_valid_index` on the whole dataframe and it skips only the nans that are present in all segments. So if segments start at different timestamps some of the segments will have NaNs.

Reply:
`DataFrame.equals` compares taking into account index.

Reply:
Ok, I made a mistake in initial solution we check timestamps after dropna. So we can't really check that sum of isna equals to true. It will give not equivalent solution.
If we want to do not-equivalent check we should discuss what kind of check do we really want here, because I'm not really sure that existing type of check is reasonable enough.

Reply:
Here we need to check the following things:
1. There are no NaNs in the datasets
2. Index of the datasets are the same